planning/mv-params.json

```diff
@@ -0,0 +1,151 @@
+the user submits something that has this format:
+
+{
+    "includes": [
+	{
+	    "name": "common",
+	    "params": [
+		{ "arg": "foo", "vals": [ "val1", "val2" ], "role": "all" },
+		{ "arg": "bar", "vals": [ "val3", "val4" ], "role": "client" }
+	    ]
+	},
+	{
+	    "name": "scenario1",
+	    "params": [
+		{ "arg": "zorg", "vals": [ "val5", "val6" ], "role": "server" },
+		{ "arg": "alpha", "vals": [ "val7", "val8" ], "role": "infra" }
+	    ]
+	},
+	{
+	    "name": "scenario2",
+	    "params": [
+		{ "arg": "beta", "vals": [ "val9", "val10" ], "role": "all" }
+	    ]
+	}
+    ],
+    "sets": [
+	[
+	    { "include": "common" },
+	    { "include": "scenario1" },
+	    { "arg": "gamma", "vals": [ "val11" ], "role": "client" }
+	],
+	[
+	    { "include": "common" },
+	    { "include": "scenario2" },
+	    { "arg": "delta", "vals": [ "val12", "val13" ], "role": "infra" }
+	],
+	[
+	    { "include": "common" },
+	    { "arg": "epsilon", "vals": [ "val14" ] }
+	]
+    ]
+}
+
+first the includes are substituted into the sets resulting in these sets:
+
+{
+    "sets":[
+	[
+	    { "arg": "foo", "vals": [ "val1", "val2" ], "role": "all" },
+	    { "arg": "bar", "vals": [ "val3", "val4" ], "role": "client" },
+	    { "arg": "zorg", "vals": [ "val5", "val6" ], "role": "server" },
+	    { "arg": "alpha", "vals": [ "val7", "val8" ], "role": "infra" },
+	    { "arg": "gamma", "vals": [ "val11" ], "role": "client" }
+	],
+	[
+	    { "arg": "foo", "vals": [ "val1", "val2" ], "role": "all" },
+	    { "arg": "bar", "vals": [ "val3", "val4" ], "role": "client" },
+	    { "arg": "beta", "vals": [ "val9", "val10" ], "role": "all" },
+	    { "arg": "delta", "vals": [ "val12", "val13" ], "role": "infra" },
+	],
+	[
+	    { "arg": "foo", "vals": [ "val1", "val2" ], "role": "all" },
+	    { "arg": "bar", "vals": [ "val3", "val4" ], "role": "client" },
+	    { "arg": "epsilon", "vals": [ "val14" ] }
+	]
+    ]
+}
+
+then the args are multiplexed within each set to create all the possible permutations:
+
+[
+    // start set 1 permutations
```

Comment:
comments are invalid, just used to document the different sections since I cheated and didn't list every single permutation since it would have been so many

Reply:
ack

Reply:
if you still want to use comments (please don't), you can use a fake _comments dataset as a workaround. I shouldn't be suggesting this because it would imply our multiplex to ignore these datasets. It is not worth it.

Reply:
maybe I wasn't clear -- the comments are just here for the sake of this conversation.  i have no intent on making comments part of the specifications/schema